wowup-electron/src/app/services/addons/addon.provider.factory.ts

```diff
@@ -68,4 +68,17 @@ export class AddonProviderFactory {
   public createWowUpAddonProvider(): WowUpAddonProvider {
     return new WowUpAddonProvider(this._httpClient, this._electronService);
   }
+
+  public getAll() : AddonProvider[] {
+    if (this._providers.length === 0) {
```

Comment:
This function replaces the list constructed in the `AddonService`. This makes sure that other places can access this list as well without causing circular references. The `WowUpService` needs this as well to set the default values.